### DISCUSSION TITLE
What's the best way to ripgrep your entire filesystem?

### DISCUSSION THREAD

**Discussion by wpcarro** [BurntSushi/ripgrep#2158] [General] [answered]
Without noisy output like:

```
proc/3680405/task/3680475/attr/keycreate: Invalid argument (os error 22)
```

**Comment by BurntSushi** [comment, marked as answer]
You can use `--no-messages`.

But, it's generally not a great idea to just run a process that touches literally every file. Some files, for example, are virtual and are _quite_ large. Try running, e.g., `wc -c /proc/1/pagemap` or something. (Well, maybe don't, because it might take quite a long time.)

What I do personally is just create a `/.ignore` file and ignore the directories that I know are likely uninteresting (or even counter productive):

```
$ cat /.ignore
/dev
/m
/media
/mnt
/proc
/root
/run
/sys
```

Then you can do things like `rg foo /`. Now, you're still quite likely to get errors, as there are probably lots of files in `/var` or even `/etc` that require root permissions to even read. That's where `--no-messages` can help.

**Reply by wpcarro** [reply]
I like this approach. Thank you 🙂
